File: ios/SmartBaibolyYarn/DailyVerseWidgetModule.m
#import <React/RCTBridgeModule.h>

// Bridge declaration for the Swift DailyVerseWidgetModule.
@interface RCT_EXTERN_MODULE(DailyVerseWidget, NSObject)
RCT_EXTERN_METHOD(refresh:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
@end

File: ios/SmartBaibolyYarn/DailyVerseWidgetModule.swift
import Foundation
import React
import WidgetKit

// Lets JS ask the home-screen widget to reload its timeline right after
// rewriting dailyVerse.json (fresh install, colour change). Same JS name and
// method as the Android module.
@objc(DailyVerseWidget)
class DailyVerseWidgetModule: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool { false }

  @objc func refresh(_ resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    WidgetCenter.shared.reloadAllTimelines()
    resolve(true)
  }
}
